From 4bfa55e5a8cae27c94bbd267fb8703a3a6cafd55 Mon Sep 17 00:00:00 2001
From: ElDainosor <eldainosor@gmail.com>
Date: Wed, 3 Apr 2019 00:07:24 -0300
Subject: [PATCH] base: Optimize toGrayscale function

 Let's transform the original bitmap before doing any changes to make it grayscale/tint it

Change-Id: Id4228db8f99f6aad85186c7544a46b53744aa5f1
Signed-off-by: Anirudh Gupta <anirudhgupta109@gmail.com>
---
 core/java/com/android/internal/util/aicp/ImageHelper.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/java/com/android/internal/util/aicp/ImageHelper.java b/core/java/com/android/internal/util/aicp/ImageHelper.java
index 9ce63f4db8c..6581a9530a8 100644
--- a/core/java/com/android/internal/util/aicp/ImageHelper.java
+++ b/core/java/com/android/internal/util/aicp/ImageHelper.java
@@ -63,6 +63,11 @@ public class ImageHelper {
         int width, height;
         height = bmpOriginal.getHeight();
         width = bmpOriginal.getWidth();
+        try {
+            bmpOriginal = RGB565toARGB888(bmpOriginal);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
 
         Bitmap bmpGrayscale = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
         Canvas c = new Canvas(bmpGrayscale);
